packages/client/src/apis/market/MarketExecuteAPI.ts

```diff
@@ -1,31 +1,27 @@
 import { BaseVertexAPI } from '../base';
 import {
+  BurnLpParams,
   CancelAndPlaceOrderParams,
   CancelOrdersParams,
   CancelProductOrdersParams,
   CancelTriggerOrdersParams,
   CancelTriggerProductOrdersParams,
-  OptionalSignatureParams,
+  MintLpParams,
   PlaceOrderParams,
   PlaceTriggerOrderParams,
 } from './types';
-import {
-  EngineExecuteBurnLpParams,
-  EngineExecuteMintLpParams,
-} from '@vertex-protocol/engine-client';
-
-import { OptionalSubaccountOwner } from '../types';
+import { OptionalSignatureParams } from '../types';
 
 export class MarketExecuteAPI extends BaseVertexAPI {
   /**
    * Mint LP tokens through engine
    * @param params
    */
-  async mintLp(params: OptionalSubaccountOwner<EngineExecuteMintLpParams>) {
+  async mintLp(params: MintLpParams) {
     return this.context.engineClient.mintLp({
       ...params,
-      chainId: await this.getSignerChainId(),
-      verifyingAddr: this.getEndpointAddress(),
+      chainId: await this.getSignerChainIdIfNeeded(params),
+      verifyingAddr: params.verifyingAddr ?? this.getEndpointAddress(),
```

Comment:
There isn't much benefit to passing this in, but I added so that we can leverage the same `OptionalSignatureParams` type